File: solutions/009/009.py
"""
Project Euler - Problem Solution 009
Problem Title - Special Pythagorean triplet
Copyright (c) Justin McGettigan. All rights reserved.
https://github.com/jwmcgettigan/project-euler-solutions
"""

def pythagorean_triplet_product(p):
  ''' Finds the product of a pythagorean
  triplet for the given perimeter (p). '''
  for a in range(1, p // 3 + 1):
    for b in range(a + 1, p // 2 + 1):
      c = p - a - b
      if (a * a + b * b == c * c):
        if(a + b + c == p):
          return a*b*c

if __name__ == "__main__":
  print(pythagorean_triplet_product(1000))

File: solutions/009/009.js
/**
 * Project Euler - Problem Solution 009
 * Copyright (c) Justin McGettigan. All rights reserved.
 * https://github.com/jwmcgettigan/project-euler-solutions
 */

/** Finds the product of a pythagorean
 * triplet for the given perimeter (p). */
const pythagoreanTripletProduct = (p) => {
  for(let a = 1; a < Math.floor(p / 3) + 1; a++) {
    for(let b = a+1; b < Math.floor(p / 2) + 1; b++) {
      k = p - i - b;
      if (a * a + b * b == k * k) {
        if(a + b + k == p) {
          return a*b*k;
        }
      }
    }
  }
};

console.log(pythagoreanTripletProduct(1000));